docs/data/data-grid/row-recipes/DetailPanelExpandCollapseAll.js

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import UnfoldLessIcon from '@mui/icons-material/UnfoldLess';
+import UnfoldMoreIcon from '@mui/icons-material/UnfoldMore';
+import {
+  DataGridPro,
+  useGridApiContext,
+  useGridSelector,
+  gridDetailPanelExpandedRowIdsSelector,
+  GRID_DETAIL_PANEL_TOGGLE_COL_DEF,
+  gridDetailPanelExpandedRowsContentCacheSelector,
+} from '@mui/x-data-grid-pro';
+import {
+  randomCreatedDate,
+  randomCurrency,
+  randomEmail,
+  randomPrice,
+} from '@mui/x-data-grid-generator';
+
+export default function DetailPanelExpandCollapseAll() {
+  const getDetailPanelContent = React.useCallback(
+    ({ row }) => <Box sx={{ p: 2 }}>{`Order #${row.id}`}</Box>,
+    [],
+  );
+
+  const getDetailPanelHeight = React.useCallback(() => 50, []);
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGridPro
+        rows={rows}
+        columns={columns}
+        getDetailPanelContent={getDetailPanelContent}
+        getDetailPanelHeight={getDetailPanelHeight}
+      />
+    </div>
+  );
+}
+
+function CustomDetailPanelHeader() {
+  const apiRef = useGridApiContext();
+  const expandedRowIds = useGridSelector(
+    apiRef,
+    gridDetailPanelExpandedRowIdsSelector,
+  );
+  const rowsWithDetailPanels = useGridSelector(
+    apiRef,
+    gridDetailPanelExpandedRowsContentCacheSelector,
+  );
+
+  // little hack to always receive the keys in the correct type
+  const allRowIdsWithDetailPanels = Object.keys(rowsWithDetailPanels).map((key) =>
+    Number.isNaN(+key) ? key : +key,
+  );
```

Comment:
What's the intention here? Converting row ids to numbers if possible?

Reply:
yep ... otherwise it won't work

Reply:
It's not a universal solution though, because if it would produce wrong results if the id is a string that also happens to be a number.
We should either use `dataRowIdToIdLookup` or alternative solutions here (see https://github.com/mui/mui-x/issues/13385)

Reply:
I did it this way because of the discussion on slack (resulting in #13385) ... I will change it to the `getRowId(dataRowIdToModelLookup[id])` suggestion to be future-proof. 👍🏼